### ISSUE TITLE
Would we consider treating data tree children as first class citizens for row operations?

### ISSUE THREAD

**Issue by BrianMcBrayer** [olifolkerd/tabulator#3574]
**Is your feature request related to a problem? Please describe.**
First, I love Tabulator. Thank you so much for this amazing library.

I am using Tabulator to show a file system (where every row is a directory or a file), and I'm using the data tree functionality to show files within directories.

Sometimes I need to update a file (`RowComponent`) that is a child. It looks like Tabulator only handles updates on top-level rows. For instance, if I ask it to scroll to a child row, or select a child row, or update a child row, it doesn't seem to be able to find it.

**Describe the solution you'd like**
It would be fantastic that, if I have the `index` of a child row, that Tabulator finds that child row and performs the action. Alternatively, if I find the child row and have the `RowComponent`, it would be neat if I could do all the operations (for my particular use-case, calling `row.update({})`) on it that I could do on a top-level row.

**Describe alternatives you've considered**
Alternatively, I need to update the parent. For deeply-nested nodes, this collapses the tree and I will need to track which child row is showing and loop down to show it.

**Comment by harbulot** [comment]
While looking into #3020, I've added a few tree functions (see comments at the bottom and examples): https://github.com/harbulot/tabulator-examples/blob/main/tree-tests-tabulator-5.1/tabulator-custom-tree-extensions.js

I guess what you're talking about is a similar workaround to what I've done in `tableTreeIndexRows`.

I don't think I've used it for tree filtering as such, but it's certainly useful to select a number of rows by ID in the tree (see `tableTreeSelectIds`). As it stands today, it's mainly been tested against Tabulator version 4.9.
